### ISSUE TITLE
PublicAPI analyzer doesn't flag addition of [Experimental] to a shipped API

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#83840]
Spotted by @jcouv in https://github.com/dotnet/roslyn/pull/83590#discussion_r3199107197.

In that PR, the C# syntax generator started emitting

```csharp
[Experimental(global::Microsoft.CodeAnalysis.RoslynExperiments.PreviewLanguageFeatureApi, UrlFormat = @"https://github.com/dotnet/roslyn/issues/82210")]
public WithElementSyntax Update(SyntaxToken withKeyword, ArgumentListSyntax argumentList)
```

on a method whose corresponding entry in `PublicAPI.Shipped.txt` is the non-experimental form

```
Microsoft.CodeAnalysis.CSharp.Syntax.WithElementSyntax.Update(...) -> Microsoft.CodeAnalysis.CSharp.Syntax.WithElementSyntax!
```

We expected the PublicAPI analyzer to flag this as a change to the declared API surface (the shipped entry is no longer accurate now that the method carries `[Experimental]`). It did not.

This means adding `[Experimental]` to a shipped, non-experimental API can land silently with no diagnostic, which defeats the point of tracking experimental status in the API files.

**Comment by 333fred** [comment]
> on a method whose corresponding entry in `PublicAPI.Shipped.txt` is the non-experimental form
>
> ```
> Microsoft.CodeAnalysis.CSharp.Syntax.WithElementSyntax.Update(...) -> Microsoft.CodeAnalysis.CSharp.Syntax.WithElementSyntax!
> ```

This is incorrect. The API was already marked experimental:

https://github.com/CyrusNajmabadi/roslyn/blob/d5f23179b8e3bdde3840d2a7c005388e5d0ec94f/src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt#L5
